docker/Dockerfile

```diff
@@ -1,32 +1,43 @@
-FROM node:lts-buster-slim
+FROM node:12-buster-slim
 
 LABEL maintainer="dave@force9.org"
 
 ENV NVM_DIR /root/.nvm
 ENV DEBIAN_FRONTEND noninteractive
-
 COPY . /enigma-bbs
 
-# Do some installing!
-RUN apt-get update && apt-get install -y \
+# Do some installing! (and alot of cleaning up) keeping it in one step for less docker layers
+# - if you need to debug i recommend to break the steps with individual RUNs)
+RUN apt-get update \
+    && apt-get install -y \
     git \
     curl \
     build-essential \
     python \
+    python 3 \
```

Comment:
This looks suspect?

Reply:
pm ERR! code 1                                                                                                                                                                                      npm ERR! path /enigma-bbs/node_modules/xxhash                                                                                                                                                        npm ERR! command failed                                                                                                                                                                              npm ERR! command sh -c node-gyp rebuild                                                                                                                                                              npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@8.3.0                                                                                                                                                               npm ERR! gyp info using node@16.13.2 | linux | x64                                                                                                                                                   npm ERR! gyp ERR! find Python                                                                                                                                                                        npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON                                                                                                                     npm ERR! gyp ERR! find Python checking if "python3" can be used                                                                                                                                      npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error                                                                                                                        npm ERR! gyp ERR! find Python checking if "python" can be used                                                                                                                                       npm ERR! gyp ERR! find Python - executable path is "/usr/bin/python"                                                                                                                                 npm ERR! gyp ERR! find Python - version is "2.7.16"
npm ERR! gyp ERR! find Python - version is 2.7.16 - should be >=3.6.0                                                                                                                                npm ERR! gyp ERR! find Python - THIS VERSION OF PYTHON IS NOT SUPPORTED                                                                                                                              npm ERR! gyp ERR! find Python                                                                                                                                                                        npm ERR! gyp ERR! find Python **********************************************************                                                                                                             npm ERR! gyp ERR! find Python You need to install the latest version of Python.                                                                                                                      npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,                                                                                                                npm ERR! gyp ERR! find Python you can try one of the following options:                                                                                                                              npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"                                                                                                                  npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)                                                                                                                                  npm ERR! gyp ERR! find Python - Set the environment variable PYTHON                                                                                                                                  npm ERR! gyp ERR! find Python - Set the npm configuration variable python:                                                                                                                           npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
npm ERR! gyp ERR! find Python For more information consult the documentation at:
npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
npm ERR! gyp ERR! find Python **********************************************************
npm ERR! gyp ERR! find Python
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
npm ERR! gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:266:16)
npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:297:7)
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:388:7)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1064:16)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
npm ERR! gyp ERR! System Linux 5.10.60-qnap
npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /enigma-bbs/node_modules/xxhash
npm ERR! gyp ERR! node -v v16.13.2
npm ERR! gyp ERR! node-gyp -v v8.3.0
npm ERR! gyp ERR! not ok

Reply:
error cleared after adding py3.

Reply:
that is a great catch thank you - this is what i get for making ad-hoc changes